p-token/src/processor/amount_to_ui_amount.rs

```diff
@@ -16,11 +16,12 @@ pub fn process_amount_to_ui_amount(
     accounts: &[AccountInfo],
     instruction_data: &[u8],
 ) -> ProgramResult {
-    let amount = u64::from_le_bytes(
-        instruction_data
-            .try_into()
-            .map_err(|_error| ProgramError::InvalidInstructionData)?,
-    );
+    let amount = if instruction_data.len() >= U64_BYTES {
+        // SAFETY: The minimum size of the instruction data is `U64_BYTES` bytes.
+        unsafe { u64::from_le_bytes(*(instruction_data.as_ptr() as *const [u8; U64_BYTES])) }
+    } else {
+        return Err(TokenError::InvalidInstruction.into());
+    };
```

Comment:
Up to you, but this could be refactored for ease of re-use between different functions